<launch>
	<group ns="fmExecutors">
		<node pkg="line_action_server" name="lineActionServer" type="lineGoalActionServer.py" output="screen">
			<param name="cmd_vel_topic" value="/fmSignals/cmd_vel" /> <!-- Topic for publishing twist messages -->
			<param name="odometry_topic" value="/odom" /> <!-- Topic for receiving odometry messages if tf is not used -->
			<param name="odom_frame" value="/odom" /> <!-- Odometry frame if tf is used -->
			<param name="base_frame" value="/odom" /> <!-- Base frame if tf is used -->
			<param name="use_tf" value="False"/> <!-- Choose tf or odometry topic -->
			
			<param name="period" value="0.1"/> <!-- sets the period of the control loop in [s] -->
			<param name="filter_size" value="10"/> <!-- Number of odom messages to average for speed estimation -->
			<param name="transition_filter_size" value="10"/> <!-- How many measurements to average over when determining zone -->
			<param name="max_linear_velocity" value="0.4"/> <!-- maximum linear velocity in autonomous mode in [m/s]-->
			<param name="max_angular_velocity" value="1.8"/> <!-- maximum angular velocity in autonomous mode in [rad/s]-->
			
			<param name="use_dynamic_reconfigure" value="False"/> <!-- If dynamic reconfigure is used, none of the settings below will have any effect -->
			
			<param name="max_distance_error" value="0.05"/> <!-- radius of the target in [m] -->

			<param name="retarder" value="0.3"/> <!-- Factor for reducing speed in difficult situations -->
					
			<param name="lin_p" value="0.4"/> <!-- Proportional gain for linear velocity controller -->
			<param name="lin_i" value="0.6"/> <!-- Integral gain for linear velocity controller -->
			<param name="lin_d" value="0.0"/> <!-- Differential gain for linear velocity controller -->
			<param name="ang_p" value="0.8"/> <!-- Proportional gain for angular velocity controller -->
			<param name="ang_i" value="0.1"/> <!-- Integral gain for angular velocity controller -->
			<param name="ang_d" value="0.05"/> <!-- Differential gain for angular velocity controller -->
			<param name="integrator_max" value="0.1"/> <!-- Common anti-windup max for integrators -->
			
			
			<param name="zone1_value" value="1"/> <!-- The transition value of zone 1 -->
	        <param name="zone1_linear_velocity" value="0.1"/> <!-- Linear velocity of the robot when it is in zone 1 [m/s] -->
	        <param name="zone1_angular_velocity" value="0.2"/> <!-- Angular velocity of the robot when it is in zone 1 [rad/s] -->
	        <param name="zone1_rabbit_factor" value="0.8"/> <!-- Placement of the aiming point on the line between robot and target -->
	        <param name="zone1_max_distance_to_line" value="0.05"/> <!-- Maximum stray distance from the line before changing zone [m] -->
	        <param name="zone1_max_angle_error" value="0.1"/> <!-- Maximum angle error before changing zone [rad] -->
	        
	        <param name="zone2_value" value="5"/> <!-- The transition value of zone 2 -->
	        <param name="zone2_linear_velocity" value="0.1"/> <!-- Linear velocity of the robot when it is in zone 2 [m/s] -->
	        <param name="zone2_angular_velocity" value="0.8"/> <!-- Angular velocity of the robot when it is in zone 2 [rad/s] -->
	        <param name="zone2_rabbit_factor" value="0.7"/> <!-- Placement of the aiming point on the line between robot and target -->
	        <param name="zone2_max_distance_to_line" value="0.25"/> <!-- Maximum stray distance from the line before changing zone [m] -->
	        <param name="zone2_max_angle_error" value="math.pi/6"/> <!-- Maximum angle error before changing zone [rad]-->
	        
	        <param name="zone3_value" value="10"/> <!-- The transition value of zone 3 -->
	        <param name="zone3_linear_velocity" value="0.1"/> <!-- Linear velocity of the robot when it is in zone 3 [m/s] -->
	        <param name="zone3_angular_velocity" value="1.2"/> <!-- Angular velocity of the robot when it is in zone 3 [rad/s] -->
	        <param name="zone3_rabbit_factor" value="0.6"/> <!-- Placement of the aiming point on the line between robot and target -->
	        
	        <param name="zone4_distance_to_target" value="0.4"/> <!-- Max distance to target before changing to zone 4 -->
	        
		</node>
	</group>	
</launch>